src/main/drivers/vtx_rtc6705.c

```diff
@@ -93,42 +94,43 @@ static uint32_t reverse32(uint32_t in)
  */
 bool rtc6705IOInit(const vtxIOConfig_t *vtxIOConfig)
 {
-    busdev = &busInstance;
-
-    busdev->busdev_u.spi.csnPin = IOGetByTag(vtxIOConfig->csTag);
-
-    bool rtc6705HaveRequiredResources =
-        (busdev->busdev_u.spi.csnPin != IO_NONE);
+    static busDevice_t busInstance;
 
-#ifdef RTC6705_POWER_PIN
-    vtxPowerPin = IOGetByTag(vtxIOConfig->powerTag);
-
-    rtc6705HaveRequiredResources = rtc6705HaveRequiredResources && vtxPowerPin;
-#endif
-
-    if (!rtc6705HaveRequiredResources) {
+    IO_t csnPin = IOGetByTag(vtxIOConfig->csTag);
+    if (!csnPin) {
         return false;
     }
 
-#ifdef RTC6705_POWER_PIN
+    vtxPowerPin = IOGetByTag(vtxIOConfig->powerTag);
+
```

Comment:
what's supposed to happen when the vtxPowerPin isn't present now?  previously the code would exit as a required hardware resource wasn't present.

there's two use cases:
1. hardware with vtx power (on/off) control.
2. hardware with an always-on vtx.


should there not be some conditional code around the lines following this?

Reply:
Not sure what should happen in this case. In the existing code if `vtxIOConfig->powerTag` isn't configured the VTX driver will not work, which I suspect might not be what is wanted, but also suggests that there is no such hardware readily available.
Do you know if there are any reason to not allow VTX devices without a power switching pin?

Reply:
Hmm, all the conditional code appears to be in the IOInit/Lo/Hi so it's probably OK in this case.

Reply:
All the SPRacing hardware has power pins for the VTX, I think the SirinFPV board doesn't have power control.

Reply:
and the SINGULARITY board doesn't either.

Reply:
the basic idea is that for boards with a power pin it MUST be defined.

Reply:
`RTC6705_POWER_PIN` is defined in any case, if that is what you mean, so the conditional does not make a difference.
And yes, all of the IO primitives are null aware. I am still tempted to wrap them inside the driver for clarity's sake - optimisation will take care of removing the double check.

Reply:
yeah, I think clarity wins in the long term.  I'm not a fan of what is essentially `null` checking inside the `IO*` methods.

To clarify what I meant, the F3NEO/F4NEO/F7DUAL/H7EXTREME all have VTX power pins.  No attempt should be made to drive the IO lines if there is no `powerTag`.  The `RTC6705_POWER_PIN` is ONLY defined in those targets.  So when RTC6705_POWER_PIN is defined, and `powerTag` is set the old code bails out.

In the case of the SINGULARITY and SIRINFPV, not having a `powerTag` is OK.

Reply:
Hmm, now I look at it, this code in `common_defaults_post.h`:

```
#ifndef RTC6705_POWER_PIN
#define RTC6705_POWER_PIN NONE
#endif
```

Is probably currently wrong (in `master`) for SIRINFPV and SINGULARITY since the `vtxPowerPin` will be `NONE` and the check for it will always fail.

Still, it doesn't really matter since what we want long-term is a solution for unified targets which needs to handle boards with and without power pins correctly.  Hmm...

Reply:
Updated to include explicit checks for `vtxPowerPin`.